File: 8_kyu/Multiple_of_index.go
package kata

func multipleOfIndex (ints []int) []int {
    accu := make([]int, len(ints))
    next := 0
    for i := 1; i < len(ints); i++ {
        if ints[i] % i == 0 {
            accu[next] = ints[i]
            next++
        }
    }
    return accu[0:next]
}

File: 8_kyu/Multiple_of_index.cs
using System;
using System.Collections.Generic;
using System.Linq;

public static class Kata {
  // Filter passes all elements which are 
  public static List <int> MultipleOfIndex(List < int > xs) =>
    xs.Where((x, i) => x == i || (i > 0 && x % i == 0))
      .ToList();
}